packages/api/src/Base.ts

```diff
@@ -374,12 +374,27 @@ export default abstract class ApiBase<CodecResult, SubscriptionResult> implement
   }
 
   private async loadMeta (): Promise<boolean> {
+    const { prebundles = {} } = this._options;
     // only load from on-chain if we are not a clone (default path), alternatively
     // just use the values from the source instance provided
     if (!this._options.source || !this._options.source._isReady) {
-      this._runtimeMetadata = await this._rpcBase.state.getMetadata();
-      this._runtimeVersion = await this._rpcBase.chain.getRuntimeVersion();
-      this._genesisHash = await this._rpcBase.chain.getBlockHash(0);
+      [this._genesisHash, this._runtimeVersion] = await Promise.all([
+        this._rpcBase.chain.getBlockHash(0),
+        this._rpcBase.chain.getRuntimeVersion()
+      ]);
+      const prebundlesKey = `${this._genesisHash}-${(this._runtimeVersion as RuntimeVersion).specVersion}`;
+      if (prebundlesKey in prebundles) {
+        try {
+          const rpcData = prebundles[prebundlesKey];
+          const metadata = new Metadata(rpcData);
+          this._runtimeMetadata = metadata;
+          this.runtimeMetadata.getUniqTypes(false);
```

Comment:
Here it needs to be true, so that this function throws an error (in case there's a mismatch), get caught, and do the RPC call

Reply:
Thanks for your comment.. May I know under what conditions we want to throw error.
I assume, if the rpcData(hex) is invalid then `new Metadata(rpcData)` will throw error and handle it. However if we have `this.runtimeMetadata.getUniqTypes(true);`, it will fail even with valid metadata with new types.  
Would like to understand the intention for this validation. Is there a case where in, genesis hash and spec version can produce two different metadata?

Reply:
Basically, the question is: if there is a type mismatch between pre-bundled metadata and types in typeRegistry, what do we do?

Solution 1: `this.runtimeMetadata.getUniqTypes(true);`, this will throw an error in the `try/catch` block, so will disregard pre-bundled, and fetch metadata from RPC call.

Solution 2: `this.runtimeMetadata.getUniqTypes(false);` this will console.error a message, but will still use pre-bundled metadata. There might be a panic further down the road, when decoding a type that is no found in typeRegistry.

I prefer solution 1: it's safer.

> Is there a case where in, genesis hash and spec version can produce two different metadata

This should be fine: genesis+specVersion should uniquely identify metadata

Reply:
If there's such type mismatch, why fetch metadata from node can make any different? Unless the result is different than the prebundled one, then the question is how? If it's by mistyping, then i believe it will fail at new Metadata(), or it could be malicious？

Reply:
> If there's such type mismatch, why fetch metadata from node can make any different?

That's the whole problem here: if we allow pre-bundled metadata, we need to trust that the runtime developer has put the correct metadata in `new Api({ prebundles })`. He can totally put a correct but outdated metadata (not malicious, but by mistake), and new Metadata() will not fail, nothing will fail on init; but some time later, something might fail (because of outdated metadata) and that will be super hard to debug.

So in my opinion there needs to be some sort of check that checks "pre-bundled metadata is compatible with node", or else it's hard-to-debug decoding errors guaranteed. I was thinking `getUniqTypes()` might be a good check, but after 2nd thought, it's not good enough. A good check would be to do `state_getMetadata` RPC call, and show warning if 2 are not same, as xlc proposed https://github.com/polkadot-js/api/issues/829#issue-430684062. But then, what's the point of having a pre-bundled metadata?

Reply:
Another solution would be to keep @karishma's current implementation (i.e. without getUniqTypes check), but just add a `console.warn` clearly stating that this is an experimental feature, and that developers should be careful with this. wdyt @jacogr ?

Reply:
Happy with the warning.

Reply:
Addressed, finally looking at the code now.